database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -25,7 +25,7 @@
  * This class implements an array-like collection on top of a Firebase location.
  */
 class FirebaseArray implements ChildEventListener {
```

Comment:
I changed everything below to be no modifier since the class itself shouldn't be accessed from anywhere outside of the package, I don't see why methods inside shouldn't follow along.